doc/shared/sunstepper/SUNStepper_Description.rst

```diff
@@ -0,0 +1,348 @@
+
+.. _SUNStepper.Description:
+
+The SUNStepper API
+==================
+
+.. versionadded:: x.y.z
+
+As with other SUNDIALS classes, the :c:type:`SUNStepper` abstract base class is
+implemented using a C structure containing a ``content`` pointer to the derived
+class member data and a structure of function pointers the derived class
+implementations of the virtual methods.
+
+.. c:type:: SUNStepper
+
+   An object for solving the IVP :eq:`SUNStepper_IVP`.
+
+   The actual definition of the ``SUNStepper`` structure is kept private to
+   allow for the object internals to change without impacting user code. The
+   following sections describe the base class methods and the virtual methods
+   that a must be provided by a derived class.
+
+.. _SUNStepper.Description.BaseMethods:
+
+Base Class Methods
+------------------
+
+This section describes methods provided by the :c:type:`SUNStepper` abstract
+base class that aid the user in implementing derived classes. This includes
+functions for creating and destroying a generic base class object, attaching and
+retrieving the derived class ``content`` pointer, and setting function pointers
+to derived class method implementations.
+
+.. _SUNStepper.Description.BaseMethods.CreateDestroy:
+
+Creating and Destroying an Object
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In additional to creating an empty :c:type:`SUNStepper` using
+:c:func:`SUNStepper_Create` described below, there is the
+:c:func:`ARKodeCreateSUNStepper` function to construct a :c:type:`SUNStepper`
+from an ARKODE integrator.
+
+.. c:function:: SUNErrCode SUNStepper_Create(SUNContext sunctx, SUNStepper *stepper)
+
+   This function creates a :c:type:`SUNStepper` object to which a user should
+   attach the member data (content) pointer and method function pointers.
+
+   :param sunctx: the SUNDIALS simulation context.
+   :param stepper: a pointer to a stepper object.
+   :return: A :c:type:`SUNErrCode` indicating success or failure.
+
+   Usage:
+
+   .. code-block:: C
+
+      /* create an instance of the base class */
+      SUNStepper stepper = NULL;
+      SUNErrCode err = SUNStepper_Create(&stepper);
+
+   .. note::
+
+      See :numref:`SUNStepper.Description.BaseMethods.Content` and
+      :numref:`SUNStepper.Description.BaseMethods.AttachFunctions`
+      for details on how to attach member data and method function pointers.
+
+
+.. c:function:: SUNErrCode SUNStepper_Destroy(SUNStepper *stepper)
+
+   This function destroys a :c:type:`SUNStepper` object.
+
+   :param stepper: a pointer to a stepper object.
+   :return: A :c:type:`SUNErrCode` indicating success or failure.
+
+   .. note::
+
+      This function only frees memory allocated within the base class and the
+      base class structure itself. The user is responsible for freeing any
+      memory allocated for the member data (content).
+
+
+.. _SUNStepper.Description.BaseMethods.SteppingFunctions:
+
+Stepping Functions
+^^^^^^^^^^^^^^^^^^
```

Comment:
Note for reviewers: Unlike `MRIStepInnerStepper`, the functions in this section are publicly documented. They can be hidden if folks think that's preferable.